model-engine/model_engine_server/domain/use_cases/sync_inference_use_cases.py

```diff
@@ -71,6 +72,14 @@ async def execute(
         await autoscaling_metrics_gateway.emit_inference_autoscaling_metric(
             endpoint_id=model_endpoint_id
         )
+        # Hack: manually resolve dns if istio is present
```

Comment:
can we add a more comprehensive explanation why we do this for our future selves